pkg/cmdutil/json_flags.go

```diff
@@ -110,6 +110,58 @@ func checkJSONFlags(cmd *cobra.Command) (*exportFormat, error) {
 	return nil, nil
 }
 
+func AddFormatFlags(cmd *cobra.Command, exportTarget *Exporter) {
+	var format string
+	StringEnumFlag(cmd, &format, "format", "", "", []string{"json"}, "Output format")
+	f := cmd.Flags()
+	f.StringP("jq", "q", "", "Filter JSON output using a jq `expression`")
+	f.StringP("template", "t", "", "Format JSON output using a Go template; see \"gh help formatting\"")
+
+	oldPreRun := cmd.PreRunE
+	cmd.PreRunE = func(c *cobra.Command, args []string) error {
+		if oldPreRun != nil {
+			if err := oldPreRun(c, args); err != nil {
+				return err
+			}
+		}
+
+		if export, err := checkFormatFlags(c); err == nil {
+			if export == nil {
+				*exportTarget = nil
+			} else {
+				*exportTarget = export
+			}
```

Comment:
```suggestion
			*exportTarget = export
```

I think this should work as expected.

Reply:
I thought that originally, but a variable that is nil is not the same as nil, it seems. I think it's because the return is `*jsonExporter` (nee `exportFormat`). It'd odd, I agree. The same goes for the [comment below](https://github.com/cli/cli/pull/7832#discussion_r1445521174) - it's related.

Reply:
Ahh makes sense 👍 Good catch.